crates/storage-query-datafusion/src/table_providers.rs

```diff
@@ -24,19 +24,18 @@ use datafusion::physical_plan::{
     DisplayAs, DisplayFormatType, ExecutionMode, ExecutionPlan, Partitioning, PlanProperties,
     SendableRecordBatchStream,
 };
-
 use restate_types::identifiers::{PartitionId, PartitionKey};
 
 use crate::context::SelectPartitions;
 use crate::table_util::compute_ordering;
 
-pub(crate) trait ScanPartition: Send + Sync + Debug + 'static {
+pub trait ScanPartition: Send + Sync + Debug + 'static {
```

Comment:
Why did you remove `(crate)`?

Reply:
Without this change, we get the following:

> warning: trait `ScanPartition` is more private than the item `RemoteScannerManager::local_partition_scanner`

I couldn't resolve that without making a whole bunch of methods including `get_partition_target_node` also `pub(crate)` so I I made this pub instead. Part of the problem is that `PartitionLocation` holds a reference to a `ScanPartition` - at least I couldn't figure out a reasonable way to get it to compile without making pub.

Reply:
Thanks for the clarification.